Assets/Scripts/SS3D/Systems/Inventory/Items/Item.cs

```diff
@@ -272,11 +273,26 @@ public bool HasTrait(Trait trait)
         /// Modify the container of this item, can pass null to make this item not depending on any container.
         /// </summary>
         [Server]
-        public void SetContainer(AttachedContainer newContainer)
+        public virtual void SetContainer(AttachedContainer newContainer)
         {
-            if (_container == newContainer)
+            IEnumerable<Collider> colliders = GetComponents<Collider>().Concat(GetComponentsInChildren<Collider>());
+
+            if ((newContainer == null) != (_container == null))
```

Comment:
I would use the xor operator ^ here, it's clearer than this condition and does the same thing.